### PR TITLE
Add Admin and Management API for TradingFees

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

**Comment by mnaichuk** [comment]
- [ ] Add `data` to TradingFee

File: spec/api/v2/admin/trading_fees_spec.rb [hunk 1]
```diff
@@ -0,0 +1,218 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+describe API::V2::Management::TradingFees, type: :request do
```
Comment (mnaichuk) [inline spec/api/v2/admin/trading_fees_spec.rb]: Change to Admin module

File: spec/api/v2/admin/trading_fees_spec.rb [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      api_post '/api/v2/admin/trading_fees/update', token: token, params: { group: 'vip-1', id: TradingFee.first.id }
+
+      expect(response).to have_http_status(201)
+      expect(JSON.parse(response.body)['maker']).to eq('0.0015')
+      expect(JSON.parse(response.body)['taker']).to eq('0.0015')
+      expect(JSON.parse(response.body)['group']).to eq('vip-1')
+      expect(JSON.parse(response.body)['market_id']).to eq('any')
+    end
+
+    it 'returns updated trading fee table with new group' do
```
Comment (mnaichuk) [inline spec/api/v2/admin/trading_fees_spec.rb]: With new market_id

File: spec/api/v2/admin/trading_fees_spec.rb [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      api_post '/api/v2/admin/trading_fees/update', token: token, params: { market_id: 'btcusd', id: TradingFee.first.id }
+
+      expect(response).to have_http_status(201)
+      expect(JSON.parse(response.body)['maker']).to eq('0.0015')
+      expect(JSON.parse(response.body)['taker']).to eq('0.0015')
+      expect(JSON.parse(response.body)['group']).to eq('any')
+      expect(JSON.parse(response.body)['market_id']).to eq('btcusd')
+    end
+
+    it 'returns updated trading fee table with new group' do
```
Comment (mnaichuk) [inline spec/api/v2/admin/trading_fees_spec.rb]: With new maker, taker fields

File: spec/api/v2/management/trading_fees_spec.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    let(:data) {}
+    let(:signers) { %i[alex jeff] }
+
+    before do
+      create(:trading_fee, maker: 0.0005, taker: 0.001, market_id: :btcusd, group: 'vip-0')
+      create(:trading_fee, maker: 0.0008, taker: 0.001, market_id: :any, group: 'vip-0')
+      create(:trading_fee, maker: 0.001, taker: 0.0012, market_id: :btcusd, group: :any)
+    end
+
+    it 'returns trading fee with btcusd market_id and vip-0 group' do
```
Comment (mnaichuk) [inline spec/api/v2/management/trading_fees_spec.rb]: It returns all trading fees tables

File: app/api/v2/admin/trading_fees.rb [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                   type: { value: BigDecimal, message: 'admin.trading_fee.non_decimal_maker' },
+                   values: { value: -> (p){ p && p >= 0 }, message: 'admin.trading_fee.invalid_maker' },
+                   desc: -> { API::V2::Admin::Entities::TradingFee.documentation[:maker][:desc] }
+          optional :taker,
+                   type: { value: BigDecimal, message: 'admin.trading_fee.non_decimal_taker' },
+                   values: { value: -> (p){ p && p >= 0 }, message: 'admin.trading_fee.invalid_taker' },
+                   desc: -> { API::V2::Admin::Entities::TradingFee.documentation[:taker][:desc] }
+          optional :group,
+                   type: String,
+                   default: ::TradingFee::ANY,
```
Comment (mnaichuk) [inline app/api/v2/admin/trading_fees.rb]: Delete this for `update` request

File: spec/api/v2/admin/trading_fees_spec.rb [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    it 'filters by group' do
+      api_get '/api/v2/admin/trading_fees', token: token, params: { group: 'vip-0' }
+
+      result = JSON.parse(response.body)
+      expect(result.map { |r| r['group'] }).to all eq 'vip-0'
+      expect(result.length).to eq TradingFee.where(group: 'vip-0').count
+    end
+  end
+
+  describe 'POST /trading_fees/new' do
```
Comment (mnaichuk) [inline spec/api/v2/admin/trading_fees_spec.rb:47]: add spec for creating new trading fees table without passing market and group id (creating default table).

**Comment by mnaichuk** [comment]
Add description for this PR please.

**Comment by mnaichuk** [comment]
Remove old columns in `admin/markets` views (show, index)

**Comment by mnaichuk** [comment]
also we need /api/v2/public/trading_fees for FE are you agree @ysv 
Or we need join market with with current trading_fee for user when you log in and when you don't just don't show fees like in https://www.binance.com/en/trade/NEO_BTC

I suggest to add `maker/taker_fee` entity for `public/markets` and use `TradingFee.def for(group:, market_id:)` for fetching right trading_fee table with maker/taker_fee values @ysv @dnfd

**Comment by ysv** [comment]
> also we need /api/v2/public/trading_fees for FE are you agree @ysv
> Or we need join market with with current trading_fee for user when you log in and when you don't just don't show fees like in https://www.binance.com/en/trade/NEO_BTC
> 
> I suggest to add `maker/taker_fee` entity for `public/markets` and use `TradingFee.def for(group:, market_id:)` for fetching right trading_fee table with maker/taker_fee values @ysv @dnfd

yes we need it

**Comment by mnaichuk** [comment]
> > also we need /api/v2/public/trading_fees for FE are you agree @ysv
> > Or we need join market with with current trading_fee for user when you log in and when you don't just don't show fees like in https://www.binance.com/en/trade/NEO_BTC
> > I suggest to add `maker/taker_fee` entity for `public/markets` and use `TradingFee.def for(group:, market_id:)` for fetching right trading_fee table with maker/taker_fee values @ysv @dnfd
> 
> yes we need it

so I will do in another patch

File: app/api/v2/admin/trading_fees.rb [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                             .eq(:group, :market_id)
+                             .build
+
+          search = TradingFee.ransack(ransack_params)
+          search.sorts = "#{params[:order_by]} #{params[:ordering]}"
+
+          present paginate(search.result), with: API::V2::Admin::Entities::TradingFee
+        end
+
+        desc 'Creates new trading fees table',
```
Comment (ysv) [inline app/api/v2/admin/trading_fees.rb]: it creates trading fees record

File: app/api/v2/admin/trading_fees.rb [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+          if trading_fee.save
+            present trading_fee, with: API::V2::Admin::Entities::TradingFee
+            status 201
+          else
+            body errors: trading_fees.errors.full_messages
+            status 422
+          end
+        end
+
+        desc 'Update trading fees table',
```
Comment (ysv) [inline app/api/v2/admin/trading_fees.rb]: it updates trading fees record

File: app/api/v2/admin/trading_fees.rb [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+          trading_fee = ::TradingFee.find(params[:id])
+          if trading_fee.update(declared(params, include_missing: false))
+            present trading_fee, with: API::V2::Admin::Entities::TradingFee
+          else
+            body errors: trading_fee.errors.full_messages
+            status 422
+          end
+        end
+
+        desc 'Deletes trading fees table',
```
Comment (ysv) [inline app/api/v2/admin/trading_fees.rb]: it deletes trading fees record

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Pull request merged**